sdk/core/azure-core/inc/credentials/credentials.hpp

```diff
@@ -38,35 +38,7 @@ namespace Azure { namespace Core { namespace Credentials {
     std::string const m_clientSecret;
 
   public:
-    ClientSecretCredential(
-        std::string const& tenantId,
-        std::string const& clientId,
-        std::string const& clientSecret)
-        : m_tenantId(tenantId), m_clientId(clientId), m_clientSecret(clientSecret)
-    {
-    }
-
-    ClientSecretCredential(
-        std::string const&& tenantId,
-        std::string const& clientId,
-        std::string const& clientSecret)
-        : m_tenantId(std::move(tenantId)), m_clientId(clientId), m_clientSecret(clientSecret)
-    {
-    }
-
-    ClientSecretCredential(
-        std::string const&& tenantId,
-        std::string const&& clientId,
-        std::string const& clientSecret)
-        : m_tenantId(std::move(tenantId)), m_clientId(std::move(clientId)),
-          m_clientSecret(clientSecret)
-    {
-    }
-
-    ClientSecretCredential(
-        std::string const&& tenantId,
-        std::string const&& clientId,
-        std::string const&& clientSecret)
+    ClientSecretCredential(std::string tenantId, std::string clientId, std::string clientSecret)
```

Comment:
I remember we have a decision that to make constructors `explicit`.

Reply:
I know, but this one has 3 arguments, is it neccessary/makes a difference?

Reply:
I don't think it's a necessary.

But that's a decision we made in C++ working group meeting.

Reply:
maybe you can also propose to revisit this decision in the next meeting. But don't just silently ignore any decision we've made.

Reply:
Ah, I remenber now - since C++11, making a multiple arg constructor explicit prevents initializer list initialization. I'll fix

Reply:
Fixed.